File: backend/tools/tool-encrypt/src/main/java/cn/edu/ntu/boot/encrypt/ToolEncryptApplication.java
package cn.edu.ntu.boot.encrypt;

import common.redis.config.ExcludeRedisConfig;
import common.swagger.annotation.EnableSwagger;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.autoconfigure.amqp.RabbitAutoConfiguration;
import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
import org.springframework.context.annotation.Import;

/**
 * @author zack <br>
 * @create 2020-05-03 21:23 <br>
 */
@EnableSwagger
@Import(ExcludeRedisConfig.class)
@SpringBootApplication(exclude = {DataSourceAutoConfiguration.class, RabbitAutoConfiguration.class})
public class ToolEncryptApplication {

    public static void main(String[] args) {
        SpringApplication.run(ToolEncryptApplication.class, args);
    }
}

File: backend/tools/tool-encrypt/src/main/kotlin/io/github/alice52/ToolEncryptApplication.kt
package io.github.alice52

import common.redis.config.ExcludeRedisConfig
import common.swagger.annotation.EnableSwagger
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.autoconfigure.amqp.RabbitAutoConfiguration
import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Import

@EnableSwagger
//@SimpleBootApplication
@Import(ExcludeRedisConfig::class)
@SpringBootApplication(exclude = [DataSourceAutoConfiguration::class, RabbitAutoConfiguration::class])
open class ToolEncryptApplication

fun main(vararg args: String) {

    runApplication<ToolEncryptApplication>(*args)
}

// logger()
inline fun <reified R : Any> R.logger(): Logger =
    LoggerFactory.getLogger(this::class.java.name.substringBefore("\$Companion"))
